### ISSUE TITLE
several tests fail locally

### ISSUE THREAD

**Issue by blondegeek** [e3nn/e3nn#125]

**Comment by blondegeek** [comment]
in `rs_test.py`: 
`test_reduce_tensor_antisymmetric_L2`
in `rsh_test.py`: 
`test_sh_cuda_single`
`TestKernelConvFn.test1`

**Comment by blondegeek** [comment]
@L-sky, can you elaborate on these issues? Especially `test_sh_cuda_single` and `TestKernelConvFn.test1` since those have to do with spherical harmonics.

**Comment by L-sky** [comment]
For a fresh conda env, both `pip install -e .` and `pip install .` result in some funky dependency error:   
[error_log_python_3_8_dependencies.txt](https://github.com/e3nn/e3nn/files/5030989/error_log_python_3_8_direct.txt)

I create env, install pytorch and pytorch_geometric, then launch tests as follows:
[installation_instructions.txt](https://github.com/e3nn/e3nn/files/5030986/installation_instructions.txt)

For envs with python 3.6, 3.7: `test_reduce_tensor_antisymmetric_L2`, `test_sh_cuda_single` and `TestKernelConvFn.test1` fail.
[error_log_python_3_7.txt](https://github.com/e3nn/e3nn/files/5030588/error_log_python_3_7.txt)

As for python 3.8, only `test_sh_cuda_single` and `TestKernelConvFn.test1` fail.
[error_log_python_3_8.txt](https://github.com/e3nn/e3nn/files/5030589/error_log_python_3_8.txt)

**Comment by L-sky** [comment]
`spherical_harmonics_alpha(l, alpha)` triggers the error when `l=1` and `alpha` is a tensor on cuda.

**Comment by L-sky** [comment]
It appears that combination of  `@torch.jit.script` and cuda can't resolve `dim=-1` for the following operation in `spherical_harmonics_alpha` (given certain shapes of sin, cos):
```
out = torch.cat([
        math.sqrt(2) * sin,
        torch.ones_like(alpha),
        math.sqrt(2) * cos
    ], dim=-1)
```

The way around it is to supply actual index of the dimension, e.g. `dim=alpha.ndim-1`

**Comment by L-sky** [comment]
For `test_reduce_tensor_antisymmetric_L2` issue comes entirely from `rs.reduce_tensor`. 
Both python 3.6 and 3.7 produce the same `Q`, for which test fails. 
Python 3.8 gives different `Q`, for which test passes.

**Comment by L-sky** [comment]
If we denote `Q` for python 3.6 as `Q_36` and for python 3.8 as `Q_38`, then `Q_38 = P @ Q_36`, 
where `P = diag([1, -1, 1, -1, 1, 1, -1, 1, 1, 1])`

**Comment by blondegeek** [comment]
Ooph! Good to know. I wonder why the signs changed...

**Comment by L-sky** [comment]
```
for i, x in enumerate(base):
    x = max(x, key=lambda xs: sum(s for s, x in xs))
    for s, e in x:
        j = full_base.index(e)
        Q[i, j] = s / len(x)**0.5
```

Disparity begins here, `base` is a set, and order of elements in a set is generally not stable. 
Although this is not necessarily the reason, it is hard for me to tell whether order is somehow accounted for later in the code.

**Comment by mariogeiger** [comment]
I reproduce the error on my machine but I still don't have any clue

**Comment by mariogeiger** [comment]
I fixed it by adding the line
```
base = sorted([sorted([sorted(xs) for xs in x]) for x in base])
```

For me it's a bug in python 3.7 because I don't see why I need to do this.
